includes/wc-update-functions.php

```diff
@@ -1409,3 +1409,41 @@ function wc_update_320_mexican_states() {
 function wc_update_320_db_version() {
 	WC_Install::update_db_version( '3.2.0' );
 }
+
+/**
+ * Synchronize purchased products database table from order meta.
+ */
+function wc_update_330_product_purchases() {
+	global $wpdb;
+
+	$statuses = array_map( 'esc_sql', wc_get_is_paid_statuses() );
+
+	// TODO: Test if this can cause a lot of performance issues during upgrade process.
```

Comment:
🤷‍♂️

Reply:
This will likely have a big impact on stores with plenty of orders. As long as we do not run this automatically and have the ability to run it as part of the WooCommerce wp-cli upgrade command I do not see an issue though.

If you have a small site you will likely not see much benefit from these changes, if you do have a large site though you would know better and then just run the upgrade via wp-cli.

We could make this clearer in the readme and develop blog post.

Reply:
@kloon, does this imply that we should have a fallback for the `WC_Product_Purchases_Helper::is_product_purchased` method?

Since if they do not run the CLI command this method will always return false.

Reply:
I do not think a fallback in needed, the migration should just be part of the DB updater script. The user will get a message to update the DB in admin, they also have the option to execute the update via wp-cli which will run the same update as what would run via the link in wp-admin.

Reply:
I will remove this TODO, seems like it's already done. The way it's currently written it will run only when the user clicks the button to update the DB, and not on upgrading WC versions (e.g. 3.2.0 -> 3.3.0).